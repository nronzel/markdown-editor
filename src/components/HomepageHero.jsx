import {
  Button,
  Flex,
  Text,
  VStack,
  useColorModeValue,
} from "@chakra-ui/react";
import React from "react";
import { NavLink } from "react-router-dom";

const HomepageHero = () => {
  return (
    <Flex
      justifyContent="space-evenly"
      alignItems="center"
      gap={7}
      h="450px"
      borderBottom="1px solid"
      borderColor={useColorModeValue("gray.300", "gray.600")}
      wrap="wrap"
    >
      <Flex alignItems="center" justifyContent="center" h="100%">
        <Text>Edit and save Markdown files right in your browser!</Text>
      </Flex>
      <Flex direction="column" gap={5} alignItems="center">
        <VStack w="100%" gap={5}>
          <Button size="lg" w="150px">
            <NavLink to="/signin">Log In</NavLink>
          </Button>
          <Button size="lg" variant="outline" w="150px" colorScheme="teal">
            <NavLink to="/signup">Sign Up</NavLink>
          </Button>
        </VStack>
      </Flex>
    </Flex>
  );
};

export default HomepageHero;
