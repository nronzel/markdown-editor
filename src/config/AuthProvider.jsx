import React, { useContext } from "react";

const AuthContext = React.createContext();

export const AuthProvider = ({ children, value }) => {
  return <AuthContext.Provider value={value}>{children}</AuthContext.Provider>;
};

export const useAuthValue = () => {
  return useContext(AuthContext);
};
