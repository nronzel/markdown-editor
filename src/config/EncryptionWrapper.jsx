import React, { useState, createContext, useContext } from "react";

export const EncryptionPasswordContext = createContext();

export const EncryptionPasswordProvider = ({ children }) => {
  const [encryptionKey, setEncryptionKey] = useState(null);
  const [isCorrectPassword, setIsCorrectPassword] = useState(false);

  async function deriveKey(password, salt) {
    const encoder = new TextEncoder();
    const passwordBuffer = encoder.encode(password);
    const saltBuffer = new Uint8Array(
      salt.match(/.{1,2}/g).map((byte) => parseInt(byte, 16))
    );
    const subtle = window.crypto.subtle;
    const key = await subtle.importKey("raw", passwordBuffer, "PBKDF2", false, [
      "deriveKey",
    ]);
    const derivedKey = await subtle.deriveKey(
      {
        name: "PBKDF2",
        salt: saltBuffer,
        iterations: 1000,
        hash: "SHA-256",
      },
      key,
      { name: "AES-GCM", length: 256 },
      false,
      ["encrypt", "decrypt"]
    );
    return derivedKey;
  }

  async function handleEncryptionKeyEntered(password, salt) {
    const derivedKey = await deriveKey(password, salt);
    setEncryptionKey(derivedKey);
    setIsCorrectPassword(true);
  }

  return (
    <EncryptionPasswordContext.Provider
      value={{
        encryptionKey,
        handleEncryptionKeyEntered,
        isCorrectPassword,
        setIsCorrectPassword,
      }}
    >
      {children}
    </EncryptionPasswordContext.Provider>
  );
};

export const useEncryptionPasswordValue = () =>
  useContext(EncryptionPasswordContext);
