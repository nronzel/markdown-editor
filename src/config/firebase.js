import { initializeApp } from "firebase/app";
import { getAuth, GoogleAuthProvider } from "firebase/auth";

const firebaseConfig = {
  apiKey: "AIzaSyD7YlbA50Vl5Qi8Qmd-LpRxwijmn7f8z2o",
  authDomain: "markdown-editor-48193.firebaseapp.com",
  projectId: "markdown-editor-48193",
  storageBucket: "markdown-editor-48193.appspot.com",
  messagingSenderId: "823260975141",
  appId: "1:823260975141:web:64d583352c28aebdc5afcc"
};

const app = initializeApp(firebaseConfig);
export const auth = getAuth(app);
export const googleProvider = new GoogleAuthProvider();
export default app
