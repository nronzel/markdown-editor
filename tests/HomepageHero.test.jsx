import { render, screen, waitFor } from "@testing-library/react";
import { vi } from "vitest";
import { ChakraProvider } from "@chakra-ui/react";
import { BrowserRouter } from "react-router-dom";
import HomepageHero from "../src/components/HomepageHero";
import { AuthProvider } from "../src/config/AuthProvider";

vi.mock("react-router-dom", () => {
  const originalModule = require("react-router-dom");
  return {
    ...originalModule,
    useNavigate: vi.fn(),
  };
});

function customRender(ui, { currentUser, ...options } = {}) {
  const Wrapper = ({ children }) => (
    <AuthProvider initialUser={currentUser}>
      <ChakraProvider>
        <BrowserRouter>{children}</BrowserRouter>
      </ChakraProvider>
    </AuthProvider>
  );

  return render(ui, { wrapper: Wrapper, ...options });
}

describe("HomepageHero", () => {
  test("renders the hero text", async () => {
    customRender(<HomepageHero />);

    await waitFor(() => {
      expect(
        screen.queryByText(/edit and save markdown files/i)
      ).toBeInTheDocument();
    });
  });

  test("renders the login section", async () => {
    customRender(<HomepageHero />);

    await waitFor(() => {
      expect(screen.queryByTestId("login-section")).toBeInTheDocument();
    });
  });

  test("matches the snapshot", async () => {
    const { container } = customRender(<HomepageHero />);

    await waitFor(() => {
      expect(container).toMatchSnapshot();
    });
  });
});
