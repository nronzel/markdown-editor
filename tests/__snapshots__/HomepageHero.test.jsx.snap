// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HomepageHero > matches the snapshot 1`] = `
<div>
  <div
    class="css-i3ugc7"
  >
    <div
      class="css-vdyfd6"
    >
      <p
        class="chakra-text css-dvpwj"
      >
        Edit and Save Markdown Files Right in Your Browser!
      </p>
      <div
        class="css-516jro"
      >
        <button
          class="chakra-button css-72u9cm"
          data-testid="animated-button"
          type="button"
        >
          Try it!
        </button>
      </div>
    </div>
    <div
      class="css-1piyg08"
      data-testid="login-section"
    >
      <p
        class="chakra-text css-157wn8n"
      >
        Sign In / Sign Up
      </p>
      <div
        class="css-krshd1"
      >
        <div
          class="css-0"
        >
          <a
            class=""
            href="/signin"
          >
            <button
              class="chakra-button css-1ozty6z"
              type="button"
            >
              <span
                class="chakra-button__icon css-1wh2kri"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-right-to-bracket "
                  data-icon="right-to-bracket"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 512 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M217.9 105.9L340.7 228.7c7.2 7.2 11.3 17.1 11.3 27.3s-4.1 20.1-11.3 27.3L217.9 406.1c-6.4 6.4-15 9.9-24 9.9c-18.7 0-33.9-15.2-33.9-33.9l0-62.1L32 320c-17.7 0-32-14.3-32-32l0-64c0-17.7 14.3-32 32-32l128 0 0-62.1c0-18.7 15.2-33.9 33.9-33.9c9 0 17.6 3.6 24 9.9zM352 416l64 0c17.7 0 32-14.3 32-32l0-256c0-17.7-14.3-32-32-32l-64 0c-17.7 0-32-14.3-32-32s14.3-32 32-32l64 0c53 0 96 43 96 96l0 256c0 53-43 96-96 96l-64 0c-17.7 0-32-14.3-32-32s14.3-32 32-32z"
                    fill="currentColor"
                  />
                </svg>
              </span>
              Log In
            </button>
          </a>
        </div>
        <div
          class="css-0"
        >
          <a
            class=""
            href="/signup"
          >
            <button
              class="chakra-button css-1gkoaai"
              type="button"
            >
              <span
                class="chakra-button__icon css-1wh2kri"
              >
                <svg
                  aria-hidden="true"
                  class="svg-inline--fa fa-user-plus "
                  data-icon="user-plus"
                  data-prefix="fas"
                  focusable="false"
                  role="img"
                  viewBox="0 0 640 512"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M96 128a128 128 0 1 1 256 0A128 128 0 1 1 96 128zM0 482.3C0 383.8 79.8 304 178.3 304h91.4C368.2 304 448 383.8 448 482.3c0 16.4-13.3 29.7-29.7 29.7H29.7C13.3 512 0 498.7 0 482.3zM504 312V248H440c-13.3 0-24-10.7-24-24s10.7-24 24-24h64V136c0-13.3 10.7-24 24-24s24 10.7 24 24v64h64c13.3 0 24 10.7 24 24s-10.7 24-24 24H552v64c0 13.3-10.7 24-24 24s-24-10.7-24-24z"
                    fill="currentColor"
                  />
                </svg>
              </span>
              Sign Up
            </button>
          </a>
        </div>
      </div>
      <p
        class="chakra-text css-qqfgvy"
      >
        Join now to access exclusive features!
      </p>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
